keps/sig-node/5526-pod-level-resource-managers/README.md

```diff
@@ -0,0 +1,1800 @@
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+A rollout could fail if a bug in the new allocation logic prevents `Guaranteed`
+pods from starting. This would not impact already running workloads, but it
+would prevent new `Guaranteed` pods from being aligned and allocated on the
+affected node. A rollback (disabling the feature gate and restarting the
+Kubelet) would mitigate this for new pods.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+A significant increase in the following metrics after enabling the feature would
+be a strong indicator that a rollback is needed:
+
+-   `topology_manager_admission_errors_total`: Indicates that the Topology
+    Manager is rejecting pods that it would have previously admitted.
+-   `cpu_manager_pinning_errors_total`: Indicates failures in the CPU allocation
+    logic.
+-   `memory_manager_pinning_errors_total`: Indicates failures in the Memory
+    allocation logic.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+Manual testing of the upgrade/downgrade path will be performed as part of the
+alpha development cycle.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+N/A.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+An operator can determine if the feature is in use by inspecting the Kubelet
+configuration on each node to see if the `PodLevelResourceManagers` feature gate
+is enabled and if the `--topology-manager-scope` is set to `pod` or `container`.
+Additionally, inspecting the logs for messages related to pod-level allocation
+would indicate usage.
```

Comment:
adding metrics may make this much simpler. Likewise, if we surface the per-container computed quasi-QOS, it became obvious when the feature is in effect for a given workload

Reply:
> adding metrics may make this much simpler.

I included the new proposed metrics as signal of the feature being used.

> Likewise, if we surface the per-container computed quasi-QOS, it became obvious when the feature is in effect for a given workload

Will follow this discussion under https://github.com/kubernetes/enhancements/pull/5527#discussion_r2372948565.